index.js

```diff
@@ -83,96 +82,87 @@ class Consumer extends EventEmitter {
       VisibilityTimeout: this.visibilityTimeout
     };
 
-    if (!this.stopped) {
-      debug('Polling for messages');
-      this.sqs.receiveMessage(receiveParams, this._handleSqsResponse);
-    } else {
+    if (this.stopped) {
       this.emit('stopped');
+      return;
     }
+    debug('Polling for messages');
+    this.sqs.receiveMessage(receiveParams)
+      .promise()
+      .then((data) => this._handleSqsResponse(data))
+      .catch((err) => {
```

Comment:
I moved the SQS error handling into this block since it is the one dealing with aws.